src/SignalR/server/SignalR/test/HubConnectionHandlerTests.cs

```diff
@@ -5329,6 +5330,155 @@ public async Task GracefulCloseDisablesReconnect()
         }
     }
 
+    [Fact]
+    public async Task HubMethodInvokesCreateActivities()
+    {
+        using (StartVerifiableLog())
+        {
+            var activities = new List<Activity>();
+            var testSource = new ActivitySource("test_source");
+            using var listener = new ActivityListener
+            {
+                ShouldListenTo = activitySource => (ReferenceEquals(activitySource, testSource))
+                    || activitySource.Name == "test_custom" || activitySource.Name == "Microsoft.AspNetCore.SignalR.Server",
```

Comment:
If your tests run in parallel then listening to sources by name is going to detect all sources with that name in the entire process, such as the sources being used by concurrently running tests. You can either disable parallelism or do these checks using reference equality to the specific ActivitySource instances used by this test.